### DISCUSSION TITLE
Showing relational data

### DISCUSSION THREAD

**Discussion by webtimal** [filamentphp/filament#2728] [Help] [answered]
What's the easiest way to show relational data inside the view page of a resource?

I have two resources ClientResource and ProjectResource which results in two Laravel relationships:

**Client model:**
```
public function projects(): HasMany
{
   return $this->hasMany(Project::class);
}
```

**Project model:**
```
public function client(): BelongsTo
{
   return $this->belongsTo(Client::class);
}
```
I have implemented a **BelongsToSelect** field inside project resource to assign a client:
```
Components\BelongsToSelect::make('client')
   ->relationship('client', 'first_name')
   ->required(),
```
Everything works fine so far, but (obviously) all I can see on the project's view page is the disabled select field showing the customer's first name. I'd like to have all related fields listed. Have I missed something crucial in the documentation or what's the best way to approach this? 

I've had a look into the RelationManager but it seems there is only a **belongsToMany** relationship (no **belongsTo**).

**Comment by danharrin** [comment, marked as answer]
You could use a virtual column in your migration to concatenate the first and last name in your database, then you can use that with your BelongsToSelect.

Alternatively, you could use a fieldset with `relationship()` to display related data in a subform, pass `->disabled()` if you dont want it editable - https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

**Reply by webtimal** [reply]
The fieldset approach was exactly what I was looking for, I wasn't aware of the `relationship()` method.
Many thanks and keep up the great work!

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/admin/resources/getting-started#relations
